lib/src/Base/Solver/LeastSquaresEquationsSolver.cxx

```diff
@@ -0,0 +1,158 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief Implementation class of an unbounded solver for systems of non-linear equations based on least square optimization
+ *
+ *  Copyright 2005-2025 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "openturns/LeastSquaresEquationsSolver.hxx"
+#include "openturns/LeastSquaresProblem.hxx"
+#include "openturns/Log.hxx"
+#include "openturns/OptimizationAlgorithm.hxx"
+#include "openturns/Log.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+
+
+BEGIN_NAMESPACE_OPENTURNS
+
+/**
+ * @class LeastSquaresEquationsSolver
+ *
+ * This class is an interface for the nonlinear LeastSquaresEquationsSolver
+ */
+
+CLASSNAMEINIT(LeastSquaresEquationsSolver)
+
+static const Factory<LeastSquaresEquationsSolver> Factory_LeastSquaresEquationsSolver;
+
+/* Parameter constructor */
+LeastSquaresEquationsSolver::LeastSquaresEquationsSolver(const Scalar absoluteError,
+                                                         const Scalar relativeError,
+                                                         const Scalar residualError,
+                                                         const UnsignedInteger maximumCallsNumber)
+  : SolverImplementation(absoluteError, relativeError, residualError, maximumCallsNumber)
+{
+  // Nothing to do
+}
+
+/* Virtual constructor */
+LeastSquaresEquationsSolver * LeastSquaresEquationsSolver::clone() const
+{
+  return new LeastSquaresEquationsSolver(*this);
+}
+
+/* String converter */
+String LeastSquaresEquationsSolver::__repr__() const
+{
+  OSS oss;
+  oss << "class=" << LeastSquaresEquationsSolver::GetClassName()
+      << " derived from " << SolverImplementation::__repr__();
+  return oss;
+}
+
+void LeastSquaresEquationsSolver::setOptimizationAlgorithm(const OptimizationAlgorithm & algorithm)
+{
+  useDefaultOptimizationAlgorithm_ = false;
```

Comment:
I dont think this flag is useful: we can just instanciate the default least squares optimizer in the ctor, see QuantileMatchingFactory

Reply:
ok will have a look, thanks